substrate/client/rpc-spec-v2/src/chain_head/tests.rs

```diff
@@ -3833,3 +3833,222 @@ async fn follow_unique_pruned_blocks() {
 	});
 	assert_eq!(event, expected);
 }
+
+#[tokio::test]
+async fn follow_report_best_block_of_a_known_block() {
+	let builder = TestClientBuilder::new();
+	let backend = builder.backend();
+	let client = Arc::new(builder.build());
+
+	let api = ChainHead::new(
+		client.clone(),
+		backend,
+		Arc::new(TaskExecutor::default()),
+		ChainHeadConfig {
+			global_max_pinned_blocks: MAX_PINNED_BLOCKS,
+			subscription_max_pinned_duration: Duration::from_secs(MAX_PINNED_SECS),
+			subscription_max_ongoing_operations: MAX_OPERATIONS,
+			operation_max_storage_items: MAX_PAGINATION_LIMIT,
+
+			max_lagging_distance: MAX_LAGGING_DISTANCE,
+			max_follow_subscriptions_per_connection: MAX_FOLLOW_SUBSCRIPTIONS_PER_CONNECTION,
+		},
+	)
+	.into_rpc();
+
+	let finalized_hash = client.info().finalized_hash;
+	let mut sub = api.subscribe_unbounded("chainHead_v1_follow", [false]).await.unwrap();
+
+	// Initialized must always be reported first.
+	let event: FollowEvent<String> = get_next_event(&mut sub).await;
+	let expected = FollowEvent::Initialized(Initialized {
+		finalized_block_hashes: vec![format!("{:?}", finalized_hash)],
+		finalized_block_runtime: None,
+		with_runtime: false,
+	});
+	assert_eq!(event, expected);
+
+	// Block tree:
+	//
+	// finalized -> block 1 -> block 2
+	//                         ^^^ best block reported
+	//
+	//           -> block 1 -> block 2_f -> block 4 -> block 5 (best)
+	//                                    ^^^ finalized
+	//
+	// The block 4 is needed on the longest chain because we want the
+	// best block 2 to be reported as pruned. Pruning is happening at
+	// height (N - 1), where N is the finalized block number.
```

Comment:
Should also not be true anymore after https://github.com/paritytech/polkadot-sdk/pull/3962

Reply:
Yep, double checked and adjusted the test, it seems we are now pruning all blocks